### PR TITLE
DI, ES run fixes and Test optimizations

### PR DESCRIPTION
**Pull request by santanusinha**

### CODE REVIEW

File: foxtrot-server/src/main/java/com/flipkart/foxtrot/server/di/FoxtrotModule.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        return environment.lifecycle()
+                .scheduledExecutorService("cardinality-executor")
+                .threads(1)
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    public List<HealthCheck> provideHealthChecks(Environment environment) {
+        return Collections.emptyList(); //TODO::REturn dropwizard healthchecks
```
Comment (r0goyal) [inline foxtrot-server/src/main/java/com/flipkart/foxtrot/server/di/FoxtrotModule.java]: ???

Comment (jitendradhawan) [reply foxtrot-server/src/main/java/com/flipkart/foxtrot/server/di/FoxtrotModule.java]: implemented hbase and hazelcast healthchecks and added provider for `List<Healthcheck>`

File: foxtrot-server/src/main/java/com/flipkart/foxtrot/server/healthcheck/HazelcastHealthCheck.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    private int counter = 0;
+
+    @Override
+    protected Result check() throws Exception {
+        // Update the counter and store in the map
+        counter = counter + 1;
+        try {
+            hazelcastConnection.getHazelcast().getMap("health_check").put(uuid, counter);
+            int toCheck = (int) hazelcastConnection.getHazelcast().getMap("health_check").get(uuid);
```
Comment (r0goyal) [inline foxtrot-server/src/main/java/com/flipkart/foxtrot/server/healthcheck/HazelcastHealthCheck.java]: move health_check to a constant

Comment (jitendradhawan) [reply foxtrot-server/src/main/java/com/flipkart/foxtrot/server/healthcheck/HazelcastHealthCheck.java]: moved to a constant

**Pull request merged**